internal/models/user.go

```diff
@@ -332,8 +341,19 @@ func (u *User) UpdatePassword(tx *storage.Connection, sessionID *uuid.UUID) erro
 }
 
 // Authenticate a user from a password
-func (u *User) Authenticate(ctx context.Context, password string) bool {
-	err := crypto.CompareHashAndPassword(ctx, u.EncryptedPassword, password)
+func (u *User) Authenticate(ctx context.Context, password string, decryptionKeys map[string]string) bool {
+	hash := u.EncryptedPassword
+
+	if es := crypto.ParseEncryptedString(u.EncryptedPassword); es != nil {
+		h, err := es.Decrypt(u.ID.String(), decryptionKeys)
```

Comment:
if there's a mix of encrypted and unencrypted passwords, will this function fail?

Reply:
Nope, `ParseEncryptedString` will return `nil` if the text contained is not an encrypted string -- a JSON (see description).